inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -920,7 +920,7 @@ layout layout_optimizer::get_expected_layout(layout const& current_layout,
 
         if (i8_u8_input) {
             if ((non_grouped || valid_grouped || valid_int8_dw) && onednn_valid_post_ops && is_2d) {
-                if (input_layout.size.batch[0] % 16 == 0) {
+                if (input_layout.size.batch[0] >= 16) {
```

Comment:
Does this improve performance? Previously, such configuration led to performance drop in some corner cases (something like `batch == 20`)

Reply:
Yes, this was a problem for network like mask-rcnn, where 2nd stage batch size is something arbitrary, like 100.
In addition, ngen is not used anymore. If the layout is single-blocked format, ocl:xe_hp is chosen. AFAIK, this is something that we should avoid..
I tested with one network and perf@b=20 was similar to perf@b=16

Reply:
I see, thank you!